mender-update/update_module/v3/platform/c++17/update_module_call.cpp

```diff
@@ -61,35 +61,57 @@ error::Error UpdateModule::StateRunner::AsyncCallState(
 		}
 	}
 
+	class OutputHandler {
+	public:
+		void operator()(const char *data, size_t size) {
+			if (size == 0) {
+				return;
+			}
+			// Get rid of exactly one trailing newline, if there is one. This is because
+			// we unconditionally print one at the end of every log line. If the string
+			// does not contain a trailing newline, add a "{...}" instead, since we
+			// cannot avoid breaking the line apart then.
+			string content(data, size);
+			if (content.back() == '\n') {
+				content.pop_back();
+			} else {
+				content.append("{...}");
+			}
+			auto lines = mender::common::SplitString(content, "\n");
+			for (auto line : lines) {
+				log::Info(prefix + line);
+			}
+		}
+		string prefix;
+	} stderr_handler {"Update Module output (stderr): "};
```

Comment:
This could be just a lambda assigned to a variable, can't it?

Reply:
Hmm, I don't think so.. I'm reusing it below with a different variable. Do you see a way to do both with just one lambda?

Reply:
Oh, I didn't see the `stdout_handler` instance. This class definition combined with an instance definition confused me.